sdgx/data_models/metadata.py

```diff
@@ -101,3 +148,42 @@ def load(cls, path: str | Path) -> "Metadata":
         path = Path(path).expanduser().resolve()
         attributes = json.load(path.open("r"))
         return Metadata().update(attributes)
+
+    def check(self):
+        """Checks column info.
+
+        When passing as input to the next module, perform necessary checks, including:
+            -Is the primary key correctly defined.
+            -Is there any missing definition of the column.
+            -Are there any  unknown columns that have been incorrectly updated.
+        """
+        # Not implemented yet
+
+        pass
+
+    def update_primary_key(self, primary_key: str | list[str], composite_primary_key: bool = False):
+        """Update the primary key of the table
+
+        When update the primary key, the original primary key will be erased.
+
+        Args:
+            primary_key(str | list[str]): the primary key or key list.
+
+            composite_primary_key(bool): whether this table use composite primary key.
+        """
+
+        if composite_primary_key is False and not isinstance(primary_key, str):
+            raise ValueError("Primary key should be a string")
+
+        if composite_primary_key is True and len(primary_key) == 0:
+            raise ValueError("Composite primary key list shoud NOT be empty.")
+
+        if composite_primary_key is True:
+            self._composite_primary_key = True
+            self.primary_key = None
+            self.primary_key_list = primary_key
+        else:
+            self._composite_primary_key = False
+            self.primary_key = primary_key
+
+        logger.info(f"Primary Key updated: {primary_key}.")
```

Comment:
And composite_primary_key seems no effort

Reply:
It is a good idea to use a single list to be compatible with a single primary key or a composite primary key. 

This problem also exists in multi-table-compbiner and will be modified together.